src/main/java/com/maddyhome/idea/vim/extension/nerdtree/AbstractDispatcher.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2003-2025 The IdeaVim authors
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+package com.maddyhome.idea.vim.extension.nerdtree
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.PlatformDataKeys
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.ui.KeyStrokeAdapter
+import com.intellij.ui.treeStructure.Tree
+import java.awt.event.KeyEvent
+import javax.swing.KeyStroke
+
+/**
+ * Handles keyboard shortcuts and delegates them to appropriate actions.
+ */
+internal abstract class AbstractDispatcher(private val mappings: Mappings) : DumbAwareAction() {
+  private val keys = mutableListOf<KeyStroke>()
+
+  override fun actionPerformed(e: AnActionEvent) {
+    var keyStroke = getKeyStroke(e) ?: return
+    val keyChar = keyStroke.keyChar
+    if (keyChar != KeyEvent.CHAR_UNDEFINED) {
+      keyStroke = KeyStroke.getKeyStroke(keyChar)
+    }
+
+    keys.add(keyStroke)
+    mappings.getAction(keys)?.let { action ->
+      val component = e.getData(PlatformDataKeys.CONTEXT_COMPONENT)
+      if (component is Tree) action.action(e, component)
+      else LOG.error("Component is not a tree: $component")
```

Comment:
Do never use `if` without brackets unless it's a super trivial one-lines. It's a matter of time until you'll add a new line after `if` and will wounder why it doesn't work. Also, it's easier to miss the `else` without the brackets.

Reply:
I see. You've mentioned this before, but I thought this piece of code could be considered "one-line" at that time. Now it seems that only simple ones like `break` will do.